src/payment/bolt11.rs

```diff
@@ -408,13 +408,26 @@ impl Bolt11Payment {
 			..PaymentDetailsUpdate::new(payment_id)
 		};
 
-		if !self.payment_store.update(&update)? {
-			log_error!(
-				self.logger,
-				"Failed to manually fail unknown payment with hash: {}",
-				payment_hash
-			);
-			return Err(Error::InvalidPaymentHash);
+		match self.payment_store.update(&update) {
+			Ok(PaymentStoreUpdateResult::EntryUpdated)
+			| Ok(PaymentStoreUpdateResult::EntryUnchanged) => (),
```

Comment:
Does the updated behavior need to be api-documented somewhere, or wasn't there a release that contained the changed logic?